frontend/src/metabase-lib/expressions/types.ts

```diff
@@ -30,3 +30,5 @@ export type MBQLClauseFunctionConfig = {
   requiresFeature?: string;
 };
 export type MBQLClauseMap = Record<string, MBQLClauseFunctionConfig>;
+
+export type ErrorWithMessage = { message: string; pos?: number; len?: number };
```

Comment:
`CompileError` says `pos` and `len` can be `null`.

Are these attributes ever used?

We can avoid at least 1 cast with this:

```suggestion
export type ErrorWithMessage = Pick<Error, 'message'>;
```

Reply:
hmm what about `pos` and `len`? actually I copied this type

Reply:
Right, they're still used in `ExpressionEditorTextfield`.

But they can still be `null`:

```suggestion
export type ErrorWithMessage = { message: string; pos?: number | null; len?: number | null };
```

This allows to remove cast from line 115 in `diagnostics.ts`.

Reply:
added `number | null`